### PR TITLE
Add a relay mechanism in the governor

### PR DESCRIPTION
**Pull request by Amxx**
#### PR Checklist
- [ ] Tests
- [ ] Documentation
- [ ] Changelog entry

### CODE REVIEW

File: contracts/governance/Governor.sol [hunk 1]
```diff
@@ -347,6 +347,18 @@ abstract contract Governor is Context, ERC165, EIP712, IGovernor {
         return weight;
     }
 
+    /**
+     * @dev Relaying mechanism allowing the acting governor (possibly a timelock) to extract value and token sent to
```
Comment (frangio) [inline contracts/governance/Governor.sol:351]:
```suggestion
     * @dev Relaying mechanism allowing the acting governor (possibly a timelock) to extract value and tokens sent to
```

File: contracts/governance/Governor.sol [hunk 2]
```diff
@@ -347,6 +347,18 @@ abstract contract Governor is Context, ERC165, EIP712, IGovernor {
...
+     * the governor by mistake.
```
Comment (frangio) [inline contracts/governance/Governor.sol:352]:
```suggestion
     * @dev Relays a transaction or function call to an arbitrary target. In cases where the governance executor
     * is some contract other than the governor itself, like when using a timelock, this function can be invoked
     * in a governance proposal to recover tokens or Ether that was sent to the governor itself by mistake.
     * Note that if the executor is simply the governor itself, use of `relay` is redundant.
```

**Comment by frangio** [comment]
Does it make sense to add a test for plain `Governor` without a timelock?

File: test/governance/extensions/GovernorTimelockCompound.test.js [hunk 3]
```diff
@@ -328,6 +328,57 @@ contract('GovernorTimelockCompound', function (accounts) {
     runGovernorWorkflow();
   });
 
+  describe('relay', function () {
```
Comment (frangio) [inline test/governance/extensions/GovernorTimelockCompound.test.js:331]: Does it make sense to add a test for plain Governor without a timelock?

Comment (Amxx) [reply test/governance/extensions/GovernorTimelockCompound.test.js:331]: The Governor without a timelock already acts by itself. using relay would add an indirection level and not add anything. So it doesn't add anything to the non-timelock flavors ... but we can still test it (and test its protected)

**Review by frangio — DISMISSED** [review]
LGTM

**Review by frangio — APPROVED** [review]

**Pull request merged**